### ISSUE TITLE
Crashed in some confitions

### ISSUE THREAD

**Issue by tim77** [johnfactotum/foliate#170]
People reporting that Foliate may crashes in some conditions, when open settings or select text. I can't reproduce this either on Xorg session and Wayland and can't post traceback for debug. Only [update for Fedora 31 (pre-release) in Bodhi](https://bodhi.fedoraproject.org/updates/FEDORA-2019-9a1684f3c7).

Maybe this broken EPUB file? :thinking:

**Comment by paulcarroty** [comment]
Tested on 2-3 files, it's definitely not the bad `.epub`. The logs: 

```
$ com.github.johnfactotum.Foliate
Gjs-Message: 13:57:40.328: JS WARNING: [resource:///com/github/johnfactotum/Foliate/js/main.js 72]: test for equality (==) mistyped as assignment (=)?

Gjs-Message: 13:57:41.298: JS WARNING: [resource:///com/github/johnfactotum/Foliate/js/main.js 2406]: reference to undefined property "isFullscreen"

(com.github.johnfactotum.Foliate:4973): GLib-GObject-CRITICAL **: 13:57:41.406: g_param_value_validate: assertion 'G_IS_PARAM_SPEC (pspec)' failed

(com.github.johnfactotum.Foliate:4973): Gtk-CRITICAL **: 13:57:41.406: gtk_container_child_notify_by_pspec: assertion 'G_IS_PARAM_SPEC (pspec)' failed

(com.github.johnfactotum.Foliate:4973): GLib-GObject-WARNING **: 13:57:41.406: ../gobject/gsignal.c:3381: invalid unclassed param spec pointer for value type 'GParam'

(com.github.johnfactotum.Foliate:4973): GLib-GObject-CRITICAL **: 13:57:41.406: g_param_spec_unref: assertion 'G_IS_PARAM_SPEC (pspec)' failed
Gjs-Message: 13:57:41.604: JS WARNING: [resource:///com/github/johnfactotum/Foliate/js/main.js 2127]: reference to undefined property "atStart"

free(): invalid pointer
Aborted (core dumped)
```

**Comment by johnfactotum** [comment]
Weird. Looks like it crashes whenever GtkStack is used. Possibly not our bug?

Does it crash if you save the following as `aboutDialog.js` and run `gjs aboutDialog.js`?

```javascript
#!/usr/bin/gjs

imports.gi.versions.Gtk = '3.0'
const { Gtk } = imports.gi

Gtk.init(null)

const aboutDialog = new Gtk.AboutDialog({
    authors: ['John Factotum'],
    artists: ['John Factotum'],
    program_name: 'Foliate',
    comments: 'A simple and modern eBook viewer',
    logo_icon_name: 'com.github.johnfactotum.Foliate',
    version: '1.5.3',
    license_type: Gtk.License.GPL_3_0,
    website: 'https://johnfactotum.github.io/foliate/',
})
aboutDialog.show()

Gtk.main()
```

**Comment by paulcarroty** [comment]
@johnfactotum works fine, even no warnings.

**Comment by johnfactotum** [comment]
@paulcarroty Does this happen with 1.5.1 and earlier? Can you try installing the older versions (or commits) to figure out the exact or approximate point where this starts happening?

**Comment by paulcarroty** [comment]
@johnfactotum found the problem: the gjs update from `1.57.90` to any newest version.

**Comment by johnfactotum** [comment]
Tested with gjs 1.58.0.

It only happens when the sidebar is enabled, which is probably why @tim77 and others cannot reproduce it. It even crashes when I try to open the GTK Inspector. The crash seems to be caused by setting the child property `icon-name` on the GtkStack items.

It seems to me that is is not our bug. As a temporary workaround you can try disabling sidebar mode.

**Comment by johnfactotum** [comment]
Seems to be a GJS problem. Code to reproduce:
```javascript
#!/usr/bin/gjs

imports.gi.versions.Gtk = '3.0'
const { Gtk } = imports.gi

Gtk.init(null)

const win = new Gtk.Window()
const stack = new Gtk.Stack()
const stackSwitcher = new Gtk.StackSwitcher({ stack })

const foo = new Gtk.Button({ label: 'foo', tooltip_text: 'Crash me' })
foo.connect('clicked', () => {
    const aboutDialog = new Gtk.AboutDialog()
    aboutDialog.show()
})
const bar = new Gtk.Label({ label: 'bar' })

stack.add_titled(foo, 'foo', 'foo')
stack.add_titled(bar, 'bar', 'bar')
stack.child_set_property(foo, 'icon-name', 'view-list-symbolic')
stack.child_set_property(bar, 'icon-name', 'user-bookmarks-symbolic')

const vbox = new Gtk.Box({ orientation: Gtk.Orientation.VERTICAL })
win.add(vbox)
vbox.pack_start(stackSwitcher, false, true, 0)
vbox.pack_start(stack, false, true, 0)
win.show_all()
win.connect('destroy', Gtk.main_quit)

Gtk.main()
```

Clicking on "foo" will crash.

On the other hand, the equivalent in Python works fine:

```python
#!/usr/bin/env python3
import gi
from gi.repository import Gtk

win = Gtk.Window()
stack = Gtk.Stack()
stackSwitcher = Gtk.StackSwitcher(stack=stack)

foo = Gtk.Button(label='foo', tooltip_text='Nothing will happen!')
def showDialog(_):
    aboutDialog = Gtk.AboutDialog()
    aboutDialog.show()
foo.connect('clicked', showDialog)

bar = Gtk.Label(label='bar')

stack.add_titled(foo, 'foo', 'foo')
stack.add_titled(bar, 'bar', 'bar')
stack.child_set_property(foo, 'icon-name', 'view-list-symbolic')
stack.child_set_property(bar, 'icon-name', 'user-bookmarks-symbolic')

vbox = Gtk.Box(orientation=Gtk.Orientation.VERTICAL)
win.add(vbox)
vbox.pack_start(stackSwitcher, False, True, 0)
vbox.pack_start(stack, False, True, 0)

win.connect('destroy', Gtk.main_quit)
win.show_all()
Gtk.main()
```

**Comment by kalev** [comment]
Can you file this against gjs, please? https://gitlab.gnome.org/GNOME/gjs/issues

**Comment by paulcarroty** [comment]
`gsettings set com.github.johnfactotum.Foliate use-sidebar false` workaround works fine.

**Comment by johnfactotum** [comment]
Issue filed: https://gitlab.gnome.org/GNOME/gjs/issues/284

**Comment by kalev** [comment]
Thanks!

**Comment by johnfactotum** [comment]
This has been fixed in gjs 1.58.1.
